pkg/api/v1/conversion_test.go

```diff
@@ -222,3 +229,87 @@ func TestResourceListConversion(t *testing.T) {
 		}
 	}
 }
+
+func TestReplicationControllerConversion(t *testing.T) {
+	// If we start with a RC, we should always have round-trip fidelity.
+	inputs := []*v1.ReplicationController{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "name",
+				Namespace: "namespace",
+			},
+			Spec: v1.ReplicationControllerSpec{
+				Replicas:        utilpointer.Int32Ptr(1),
+				MinReadySeconds: 32,
+				Selector:        map[string]string{"foo": "bar", "bar": "foo"},
+				Template: &v1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{"foo": "bar", "bar": "foo"},
+					},
+					Spec: v1.PodSpec{
+						Containers: []v1.Container{
+							{
+								Name:  "container",
+								Image: "image",
+							},
+						},
+					},
+				},
+			},
+			Status: v1.ReplicationControllerStatus{
+				Replicas:             1,
+				FullyLabeledReplicas: 2,
+				ReadyReplicas:        3,
+				AvailableReplicas:    4,
+				ObservedGeneration:   5,
+				Conditions: []v1.ReplicationControllerCondition{
+					{
+						Type:               v1.ReplicationControllerReplicaFailure,
+						Status:             v1.ConditionTrue,
+						LastTransitionTime: metav1.NewTime(time.Unix(123456789, 0)),
+						Reason:             "Reason",
+						Message:            "Message",
+					},
+				},
+			},
+		},
+	}
+
+	// Add some fuzzed RCs.
+	apiObjectFuzzer := fuzzer.FuzzerFor(kapitesting.FuzzerFuncs, rand.NewSource(152), api.Codecs)
+	for i := 0; i < 10; i++ {
+		rc := &v1.ReplicationController{}
+		apiObjectFuzzer.Fuzz(rc)
+		rc.Spec.Template = &v1.PodTemplateSpec{}
+		apiObjectFuzzer.Fuzz(rc.Spec.Template)
+		for j := 0; j < 3; j++ {
+			cond := &v1.ReplicationControllerCondition{}
+			apiObjectFuzzer.Fuzz(cond)
+			rc.Status.Conditions = append(rc.Status.Conditions, *cond)
+		}
+		inputs = append(inputs, rc)
+	}
+
+	// Round-trip the input RCs before converting to RS.
+	for i := range inputs {
+		inputs[i] = roundTrip(t, inputs[i]).(*v1.ReplicationController)
+	}
+
+	for _, in := range inputs {
+		rs := &extensions.ReplicaSet{}
+		if err := k8s_api_v1.Convert_v1_ReplicationController_to_extensions_ReplicaSet(in, rs, nil); err != nil {
```

Comment:
this conversion is unsafe (can end up sharing memory on some objects)... should either call a top-level `api.Scheme.Convert` or call convert with a deep copy of `in`